source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -998,6 +998,338 @@ See the `Load Balancer Specification <../load-balancers/load-balancers.rst#event
+       as defined in `Events <#events>`_ above:
+
+       - Stale: "Connection became stale because the pool was cleared"
+       - Idle: "Connection has been available but unused for longer than the configured max idle time"
+       - Error: "An error occurred while using the connection"
+       - Pool closed: "Connection pool was closed" 
+
+The unstructured form SHOULD be as follows, using the values defined in the structured format above to fill in placeholders as appropriate:
+
+  Connection closed with driver-generated ID {{driverConnectionId}} to {{serverHost}}:{{serverPort}}. Reason: {{reason}}
+
+Connection Checkout Started Message
+-----------------------------------
+In addition to the common fields defined above, this message MUST contain the following key-value pairs:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1
+
+   * - Key
+     - Suggested Type
+     - Value
+
+   * - message
+     - String
+     - "Connection checkout started"
+
+The unstructured form SHOULD be as follows, using the values defined in the structured format above to fill in placeholders as appropriate:
+
+  Checkout started for connection to {{serverHost}}:{{serverPort}}
+
+Connection Checkout Failed Message
+-----------------------------------
+In addition to the common fields defined above, this message MUST contain the following key-value pairs:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1
+
+   * - Key
+     - Suggested Type
+     - Value
+
+   * - message
+     - String
+     - "Connection checkout failed"
+
+   * - reason
+     - String
+     - A string describing the reason checkout. The following strings MUST be used for each possible reason
+       as defined in `Events <#events>`_ above:
+
+       - Timeout: "Wait queue timeout elapsed without a connection becoming available"
+       - ConnectionError: "An error occurred while trying to establish a new connection"
+       - Pool closed: "Connection pool was closed"
+
+The unstructured form SHOULD be as follows, using the values defined in the structured format above to fill in placeholders as appropriate:
+
+  Checkout failed for connection to {{serverHost}}:{{serverPort}}. Reason: {{reason}}
+
+Connection Checked Out
+-----------------------
+In addition to the common fields defined above, this message MUST contain the following key-value pairs:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1
+
+   * - Key
+     - Suggested Type
+     - Value
+
+   * - message
+     - String
+     - "Connection checked out"
+
+   * - driverConnectionId
+     - Int
+     - The driver-generated ID for the connection as defined in `Connection <#connection>`_.
+
+The unstructured form SHOULD be as follows, using the values defined in the structured format above to fill in placeholders as appropriate:
+
+  Connection checked out with driver-generated ID {{driverConnectionId}} to {{serverHost}}:{{serverPort}}
```

Comment:
This is kind of a grammar nit, but I think this message reads a bit easier if the "with id" and "to \<address\>" phrases came before the verb here, since they're describing "Connection". If you drop the "with" part, the issue is a bit more obvious (e.g. "Connection checked out to a:12345" doesn't quite make sense).

Something like "Connection to a:12345 with driver-generated ID checked out" reads a bit better to me. The drawback being "Connection" and "checked out" are now separate, though I'm not sure if that's an issue or not. A more dramatic shift that doesn't have either problem could be something like "Connection checked out (driver-generated ID=1, destination=a:12345)".

WDYT?

Reply:
yeah I was having a hard time figuring out how to write the english for some of these and it shows here 😅 

re-reading this I had a couple thoughts:
1) I feel like the host/port should come before the ID as it feels more important given that IDs are only unique per-pool/endpoint
2) maybe "for" would be a better word than to?

I do think keeping the "Connection checked out" part together and at the start of the message is useful so that if you are skimming you can easily see what the message is about from the start of it.

given these WDYT of either of the following:
1) Connection checked out for endpoint a:12345 with driver-generated ID 123
2) Connection checked out: endpoint=a:12345, driver-generated ID=123.

(borrowing "endpoint" terminology from elsewhere in CMAP)

Reply:
I kinda like 2. It's consistent and pretty easy to parse visually (and programatically in case a given language can't do structured logs).

"endpoint" is spec language-y though,  I'm not sure if it ends up in our API or docs that often. What about just using "address" like the events do?

Reply:
cool, I updated to the second one (but with address instead of endpoint) and updated some of the other log messages to make the format/phrasing more consistent